model/models.py

```diff
@@ -94,25 +105,25 @@ def __init__(self,
             tf.TensorSpec(shape=(None, None, None, None), dtype=tf.float32),
         ]
         self.debug = debug
-        self.__apply_all_signatures()
+        self._apply_all_signatures()
     
     @property
     def step(self):
         return int(self.optimizer.iterations)
     
-    def __apply_all_signatures(self):
-        self.forward = self.__apply_signature(self._forward, self.forward_input_signature)
-        self.train_step = self.__apply_signature(self._train_step, self.training_input_signature)
-        self.val_step = self.__apply_signature(self._val_step, self.training_input_signature)
-        self.forward_encoder = self.__apply_signature(self._forward_encoder, self.encoder_signature)
-        self.forward_decoder = self.__apply_signature(self._forward_decoder, self.decoder_signature)
-    
     def __apply_signature(self, function, signature):
```

Comment:
I would not use double underscore here. Generally its seen not pythonic since its hard to mess with it later.